src/robot/client.hpp

```diff
@@ -81,29 +95,53 @@ class RobotClient {
         auto resource = this->resource_by_name(std::move(r));
         return std::dynamic_pointer_cast<T>(resource);
     }
+
+    /// @brief Get the configuration of the frame system of the given robot.
+    /// @return The configuration of the calling robot's frame system.
     std::vector<FrameSystemConfig> get_frame_system_config(
         std::vector<Transform> additional_transforms = std::vector<Transform>());
+
+    /// @brief Get the list of operations currently running on a robot.
+    /// @return The list of operations currently running on the calling robot.
     std::vector<viam::robot::v1::Operation> get_operations();
+
+    /// @brief Get the status of the robot's components.
+    /// @param components An optional list of the specific components for which status is desired.
+    /// @return A list of statuses.
     std::vector<Status> get_status(
         std::vector<ResourceName> components = std::vector<ResourceName>());
     std::vector<viam::robot::v1::Discovery> discover_components(
         std::vector<viam::robot::v1::DiscoveryQuery> queries);
 
-    std::vector<std::shared_ptr<std::thread>> threads;
-
+    /// @brief Transform a given `Pose` to a new specified destination which is a reference frame.
+    /// @param query The pose that should be transformed.
+    /// @param destination The name of the reference frame to transform the given pose to.
+    /// @return the `PoseInFrame` of the transformed pose.
     viam::common::v1::PoseInFrame transform_pose(
         viam::common::v1::PoseInFrame query,
         std::string destination,
         std::vector<Transform> additional_transforms = std::vector<Transform>());
+
+    /// @brief Blocks on the specified operation of the robot, returning when it is complete.
+    /// @param id The ID of the operation to block on.
     void block_for_operation(std::string id);
+
+    /// @brief Cancel all operations for the robot and stop all actuators and movement.
     void stop_all();
+
+    /// @brief Cancel all operations for the robot and stop all actuators and movement.
+    /// @param extra Any extra params to pass to resources' `stop` methods, keyed by `ResourceName`.
     void stop_all(std::unordered_map<ResourceName,
                                      std::unordered_map<std::string, std::shared_ptr<ProtoType>>,
                                      ResourceNameHasher,
                                      ResourceNameEqual> extra);
+
+    /// @brief Cancel a specified operation on the robot.
+    /// @param id The ID of the operation to cancel.
     void cancel_operation(std::string id);
 
    private:
+    std::vector<std::shared_ptr<std::thread>> threads_;
```

Comment:
(driveby) no one outside of robot client was actually using `threads` so it was trivial to make it private.